cpp/src/groupby/groupby.cu

```diff
@@ -146,7 +146,8 @@ groupby::groups groupby::get_groups(table_view values, rmm::mr::device_memory_re
 
   std::unique_ptr<table> grouped_values{nullptr};
   if (values.num_columns()) {
-    grouped_values = cudf::experimental::detail::gather(values, helper().key_sort_order());
+    grouped_values = cudf::experimental::detail::gather(
+      values, helper().key_sort_order(), false, false, false, mr);
```

Comment:
This is kind of API that advocate trying to avoid using scoped enumerators.
```cpp
false, false, false
```
Looks like this is pre-existing - I can set up an issue to address this in a follow PR. 
```cpp
std::unique_ptr<table> gather(table_view const& source_table,
                              column_view const& gather_map,
                              bool check_bounds,
                              bool ignore_out_of_bounds,
                              bool allow_negative_indices,
                              rmm::mr::device_memory_resource* mr,
                              cudaStream_t stream)
```

Reply:
Yeah, it felt icky adding these parameters to the call.